test-resources/template.yaml

```diff
@@ -0,0 +1,142 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Transform: AWS::Serverless-2016-10-31
+
+Parameters:
+  Environment:
+    Description: The environment type
+    Type: String
+    Default: dev
+    AllowedValues:
+      - dev
+      - build
+
+  PermissionsBoundary:
+    Description: |
+      The ARN of the permissions boundary to apply to any role created by the template
+    Type: String
+    Default: none
+
+  BackendApi:
+    Description: The Stack name of the Async Backend stack
+    Type: String
+    Default: async-backend-api
+
+Conditions:
+  UsePermissionsBoundary: !Not
+    - !Equals
+      - !Ref PermissionsBoundary
+      - none
+
+  IsDevOrBuild: !Or
+    - !Equals
+      - !Ref Environment
+      - dev
+    - !Equals
+      - !Ref Environment
+      - build
+
+Resources:
+  # Dequeue resources
+  DequeueFunction:
+    Type: AWS::Serverless::Function
+    Condition: IsDevOrBuild
+    DependsOn:
+      - DequeueLogGroup
+    Metadata:
+      BuildMethod: esbuild
+      BuildProperties:
+        Minify: true
+        Target: es2022
+        Sourcemap: false
+        EntryPoints:
+          - dequeue/dequeueHandler.ts
+    Properties:
+      FunctionName: !Sub ${AWS::StackName}-dequeue
+      Runtime: nodejs20.x
+      Handler: testDequeueHandler.lambdaHandler
```

Comment:
is this correct still?

Reply:
Sorry I thought I changed that. Nice spot! Thank you